lib/SparkPost/MessageEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace SparkPost;
+
+/**
+ * SDK class for querying the Message Events API
+ *
+ * @see https://developers.sparkpost.com/api/#/reference/message-events
+ */
+class MessageEvent extends APIResource
+{
+  public $endpoint = 'message-events';
+
+  /**
+   * Method for issuing search requests to the Message Events API.
+   *
+   * The method passes-through all of the query parameters - the valid ones are listed at
+   * @link https://developers.sparkpost.com/api/#/reference/message-events/events-documentation/search-for-message-events
+   *
+   * @param array $queryParams  The query parameters.  Note that a query parameter containing an array
+   * is collapsed into a comma-separated list.
+   *
+   * @return array The result of the query.
+   */
+  public function search(Array $queryParams)
+  {
+    // check for DateTime objects & replace them with the formatted string equivalent
+    foreach(["from", "to"] as $dateTimeParam) {
+      if (isset($queryParams[$dateTimeParam]) && $queryParams[$dateTimeParam] instanceof \DateTime) {
+        // the message events API doesn't allow the seconds or GMT offset, so strip them
+        $queryParams[$dateTimeParam] = substr($queryParams[$dateTimeParam]->format(\DateTime::ATOM), 0, 16);
+      }
+    }
```

Comment:
Should this functionality be moved into the base class like the list collapse? Thoughts?